Dalamud/Game/Gui/ChatGui2.cs

```diff
@@ -0,0 +1,580 @@
+    /// <param name="source">The source of the message (game, Dalamud, or plugin).</param>
+    /// <param name="sourceName">The name of the message source.  This is the plugin name when the source is a plugin.</param>
+    public delegate void OnMessageUnhandledDelegate(XivChatType2 type, uint timestamp, SeString sender, SeString message, XivChatMessageSource source, string sourceName);
+
+    [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+    private delegate IntPtr PrintMessageDelegate(IntPtr manager, XivChatType2 chatType, IntPtr senderName, IntPtr message, uint timestamp, IntPtr parameter);
+
+    [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+    private delegate void PopulateItemLinkDelegate(IntPtr linkObjectPtr, IntPtr itemInfoPtr);
+
+    [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+    private delegate void InteractableLinkClickedDelegate(IntPtr managerPtr, IntPtr messagePtr);
+
+    /// <summary>
+    /// Event that will be fired when a chat message is sent to chat by the game.
+    /// </summary>
+    public event OnMessageDelegate ChatMessage;
+
+    /// <summary>
+    /// Event that allows you to stop messages from appearing in chat by setting the isHandled parameter to true.
+    /// </summary>
+    public event OnCheckMessageHandledDelegate CheckMessageHandled;
+
+    /// <summary>
+    /// Event that will be fired when a chat message is handled by Dalamud or a Plugin.
+    /// </summary>
+    public event OnMessageHandledDelegate ChatMessageHandled;
+
+    /// <summary>
+    /// Event that will be fired when a chat message is not handled by Dalamud or a Plugin.
+    /// </summary>
+    public event OnMessageUnhandledDelegate ChatMessageUnhandled;
+
+    /// <summary>
+    /// Gets the ID of the last linked item.
+    /// </summary>
+    public int LastLinkedItemId { get; private set; }
+
+    /// <summary>
+    /// Gets the flags of the last linked item.
+    /// </summary>
+    public byte LastLinkedItemFlags { get; private set; }
+
+    /// <summary>
+    /// Dispose of managed and unmanaged resources.
+    /// </summary>
+    void IDisposable.Dispose()
+    {
+        this.printMessageHook.Dispose();
+        this.populateItemLinkHook.Dispose();
+        this.interactableLinkClickedHook.Dispose();
+    }
+
+    /// <summary>
+    /// Queue a chat message. While method is named as PrintChat, it only adds an entry to the queue,
+    /// later to be processed when UpdateQueue() is called.
+    /// </summary>
+    /// <param name="chat">A message to send.</param>
+    /// <param name="pluginInterface">Your plugin's <see cref="DalamudPluginInterface"/>.</param>
+    /// <param name="messagePrintedCallback">A delegate that is invoked once the message is actually printed to the chat log.  The parameter is the RaptureLogModule message index.</param>
+    public void PrintChat(XivChatEntry2 chat, DalamudPluginInterface pluginInterface, Action<nint> messagePrintedCallback = null)
```

Comment:
You should get the calling plugin via `FindCallingPlugin()` in `PluginManager` instead. If possible, inject it as a dependency into this.